distributed/tests/py3_test_client.py

```diff
@@ -37,3 +37,21 @@ def test_as_completed_async_for(c, s, a, b):
     yield f()
 
     assert set(results) == set(range(1, 11))
+
+
+def test_async_with(loop):
+    results = []
```

Comment:
Since this is a Python 3-only test file, you don't need these tricks and can simply use `nonlocal` variables instead.